src/adapters/scene_category.js

```diff
@@ -1,21 +1,66 @@
-import { Marker } from 'mapbox-gl--ENV';
 import constants from '../../config/constants.yml';
-import { createIcon } from '../adapters/icon_manager';
 import Telemetry from 'src/libs/telemetry';
 import { toUrl } from 'src/libs/pois';
 import { fire, listen } from 'src/libs/customEvents';
+import { poisToGeoJSON, emptyFeatureCollection } from 'src/libs/geojson';
+import { filteredPoisStyle, hoveredPoiStyle } from 'src/adapters/pois_styles';
+import { createDefaultMapIcon } from 'src/adapters/icon_manager';
 
-function getMarkerId(poi) {
-  return `marker_${poi.id}`;
-}
+const DYNAMIC_POIS_LAYER = 'poi-filtered';
+const ACTIVE_POIS_LAYER = 'poi-active';
 
 export default class SceneCategory {
   constructor(map) {
     this.map = map;
-    this.markers = [];
+
+    createDefaultMapIcon().then(imageData => {
+      this.map.addImage('pin_with_dot', imageData);
+    });
+
+    this.map.addSource(DYNAMIC_POIS_LAYER, {
+      type: 'geojson',
+      data: emptyFeatureCollection,
+      promoteId: 'id', // tells MapBox-GL to use this property as internal feature identifier
```

Comment:
I don't understand mapbox's documentation about this `promoteId` property, and don't know why this is not the default behavior :confused:

Reply:
MapBox-GL really needs unique identifiers to work, so by default it doesn't trust the data source for that and has its own internal system to assign ids. I think that was the safest way for them, and it's ok to be able to override it :)

Reply:
Also the geojson standard defines an optional `id` field for the feature (outside of the `properties` object). So I think that what mapbox-gl would use by default.